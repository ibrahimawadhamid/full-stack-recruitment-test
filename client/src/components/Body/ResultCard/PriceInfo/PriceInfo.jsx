import React, {Component} from 'react';
import BpkCard from 'bpk-component-card';
import {BpkGridContainer, BpkGridRow, BpkGridColumn} from 'bpk-component-grid';
import BpkImage from 'bpk-component-image';
import STYLES from './PriceInfo.scss';
import BpkText from "bpk-component-text";
import LongArrowRightIcon from 'bpk-component-icon/lg/long-arrow-right';
import {withAlignment} from 'bpk-component-icon';
import {lineHeightLg, iconSizeLg, lineHeightXxl} from 'bpk-tokens/tokens/base.es6';
import BpkButton from 'bpk-component-button';
import * as actions from "../../../../store/actions";
import connect from "react-redux/es/connect/connect";

const AlignedArrow = withAlignment(
  LongArrowRightIcon, lineHeightXxl, iconSizeLg
);

const classes = className => STYLES[className] || 'UNKNOWN';

class PriceInfo extends Component {
  render() {
    const otherDeals = this.props.priceInfo.length > 0 ? <BpkText textStyle="xs"> other deals</BpkText> : null
    return (
      <BpkGridRow>
        <BpkGridColumn width={8}>
          <BpkGridRow>
            <BpkText textStyle="xl" className={classes('leftAlign')}>{this.props.currencySymbol}{this.props.priceInfo[0]["Price"]}</BpkText>
            {otherDeals}
          </BpkGridRow>
          <BpkGridRow>
            <BpkText textStyle="base" className={classes('leftAlign')+' '+classes('grayFill300')}>{this.props.priceInfo[0]["Agent"]}</BpkText>
          </BpkGridRow>
        </BpkGridColumn>
        <BpkGridColumn width={4}>
          <BpkButton large className={classes('rightAlign')}>Select</BpkButton>
        </BpkGridColumn>
      </BpkGridRow>
    );
  };
};


const mapStateToProps = state => {
  return {
    currencySymbol: state.searchResults["currencySymbol"],
  };
}

export default connect(mapStateToProps, null)(PriceInfo);
