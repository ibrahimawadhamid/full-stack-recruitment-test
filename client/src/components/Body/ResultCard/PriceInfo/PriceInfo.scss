@import '~bpk-mixins';

.relativePosition {
  position: relative;
}

.selectButton {
  position: absolute;
  right: 0;
  top: 1.875rem;
}
.rightAlign {
  float: right;
}

.leftAlign {
  float: left;
}

.bottomAlign {
  position: absolute;
  bottom: .375rem;
}

.bottomMargin {
  margin-bottom: 1.5rem;
}

.grayFill500 {
  color: $bpk-color-gray-500;
}
.grayFill300 {
  color: $bpk-color-gray-300;
}

.greenFill {
  color: $bpk-color-green-500;
}

.arrowGrayColor {
  fill: $bpk-color-gray-300;
}
