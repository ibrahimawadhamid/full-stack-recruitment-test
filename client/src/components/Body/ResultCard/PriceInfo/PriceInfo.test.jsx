import React from 'react';
import TestRenderer from 'react-test-renderer';

import { PriceInfo } from './PriceInfo';

describe('PriceInfo', () => {
    it('should render correctly', () => {
        const tree = TestRenderer.create(<PriceInfo priceInfo={[{
            Agent: 'agent',
            Price: 'price'
        }]} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});