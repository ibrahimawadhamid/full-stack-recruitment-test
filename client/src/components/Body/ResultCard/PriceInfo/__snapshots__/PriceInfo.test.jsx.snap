// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInfo should render correctly 1`] = `
<div
  className="bpk-grid__row bpk-grid__row--padded"
>
  <div
    className="bpk-grid__column bpk-grid__column--8 bpk-grid__column--padded"
  >
    <div
      className="bpk-grid__row bpk-grid__row--padded relativePosition"
    >
      <span
        className="bpk-text bpk-text--xl leftAlign"
      >
        price
      </span>
    </div>
    <div
      className="bpk-grid__row bpk-grid__row--padded"
    >
      <span
        className="bpk-text bpk-text--base leftAlign grayFill300"
      >
        agent
      </span>
    </div>
  </div>
  <div
    className="bpk-grid__column bpk-grid__column--4 bpk-grid__column--padded relativePosition"
  >
    <button
      className="bpk-button bpk-button--large selectButton"
      disabled={false}
      onClick={null}
      type="button"
    >
      Select
    </button>
  </div>
</div>
`;
